# ãƒ‡ã‚£ãƒ¬ã‚¯ã‚¿ãƒ¼ï¼ˆDirectorï¼‰
extends Node2D


# Called when the node enters the scene tree for the first time.
func _ready():
	$"KeyConfigArtist".entry()


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	$"KeyConfigArtist".on_process(delta)


func _unhandled_input(event):

	# èµ·å‹•ç›´å¾Œã«ã€æŠ¼ã—ã¦ã‚‚ãªã„ãƒ¬ãƒãƒ¼ãŒã€€æŠ¼ã—ãŸã“ã¨ã«ãªã£ã¦ã„ã‚‹ã“ã¨ãŒã‚ã‚‹
	var event_as_text = event.as_text()
	print("å…¥åŠ›ï¼šã€€" + event_as_text)
	
	if $"KeyConfigArtist".turn_state != &"Input":
		return

	var is_ok = false
	var acception = "å—ä»˜ï¼šã€€"

	# ğŸ“–ã€€[enum JoyButton:](https://docs.godotengine.org/en/stable/classes/class_%40globalscope.html#enum-globalscope-joybutton)
	# ãƒ¬ãƒãƒ¼ã¯ -1 ï½ 10ã€ ãƒœã‚¿ãƒ³ã¯ -1 ï½ 128 ã¾ã§ã‚ã‚‹ãã†ã 
	if not is_ok:
		var matched = $"KeyConfigArtist".re_button.search(event_as_text)
		if matched:
			$"KeyConfigArtist".button_number = int(matched.get_string(1))
			$"KeyConfigArtist".button_presentation_name = "ãƒœã‚¿ãƒ³" + str($"KeyConfigArtist".button_number)
			is_ok = true

	if not is_ok:
		var matched = $"KeyConfigArtist".re_lever.search(event_as_text)
		if matched:
			var number = int(matched.get_string(1))
			$"KeyConfigArtist".button_presentation_name = "ãƒ¬ãƒãƒ¼" + str(number)
			$"KeyConfigArtist".button_number = number + 1000
			is_ok = true

	if is_ok:
		print(acception)
		$"KeyConfigArtist".turn_state = &"InputOk"
		$"Musician/SE/ğŸ””ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°æ±ºå®šéŸ³".play()
		print("å…¥åŠ›å®Œäº†")
